nixos/modules/services/web-apps/snipe-it.nix

```diff
@@ -0,0 +1,394 @@
+              fastcgi_param REDIRECT_STATUS 200;
+              fastcgi_pass unix:${config.services.phpfpm.pools."snipe-it".socket};
+              ${optionalString (cfg.nginx.addSSL || cfg.nginx.forceSSL || cfg.nginx.onlySSL || cfg.nginx.enableACME) "fastcgi_param HTTPS on;"}
+            '';
+          };
+          "~ \.(js|css|gif|png|ico|jpg|jpeg)$" = {
+            extraConfig = "expires 365d;";
+          };
+        };
+      }];
+    };
+
+    systemd.services.snipe-it-setup = {
+      description = "Preperation tasks for snipe-it";
+      before = [ "phpfpm-snipe-it.service" ];
+      after = optional db.createLocally "mysql.service";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+        User = user;
+        WorkingDirectory = snipe-it;
+      };
+      script = ''
+        set -xeo pipefail
+        umask 077
+        # create artisan .env file
+        cat <<__EOF > "${cfg.dataDir}/.env"
+        APP_ENV=production
+        APP_KEY=base64:$(head -n1 ${cfg.appKeyFile})
+        APP_URL=https://${cfg.hostName}
+        DB_HOST=${db.host}
+        DB_PORT=${toString db.port}
+        DB_DATABASE=${db.name}
+        DB_USERNAME=${db.user}
+        ${optionalString (db.passwordFile != null) "DB_PASSWORD=$(head -n1 ${db.passwordFile})"}
+        MAIL_DRIVER=${mail.driver}
+        MAIL_FROM_NAME='${mail.from.name}'
+        MAIL_FROM_ADDR=${mail.from.address}
+        MAIL_REPLYTO_NAME='${mail.from.name}'
+        MAIL_REPLYTO_ADDR=${mail.from.address}
+        MAIL_BACKUP_NOTIFICATION_ADDRESS=${mail.backupNotificationAddress}
+        MAIL_HOST=${mail.host}
+        MAIL_PORT=${toString mail.port}
+        ${optionalString (mail.user != null) "MAIL_USERNAME=${mail.user}"}
+        ${optionalString (mail.passwordFile != null) "MAIL_PASSWORD=$(head -n1 ${mail.passwordFile})"}
+        ${optionalString (mail.encryption != null) "MAIL_ENCRYPTION=${mail.encryption}"}
+        APP_SERVICES_CACHE=${cfg.cacheDir}/services.php
+        APP_PACKAGES_CACHE=${cfg.cacheDir}/packages.php
+        APP_CONFIG_CACHE=${cfg.cacheDir}/config.php
+        APP_ROUTES_CACHE=${cfg.cacheDir}/routes-v7.php
+        APP_EVENTS_CACHE=${cfg.cacheDir}/events.php
+        ${optionalString (cfg.nginx.addSSL || cfg.nginx.forceSSL || cfg.nginx.onlySSL || cfg.nginx.enableACME) "SESSION_SECURE_COOKIE=true"}
+        ${toString cfg.extraConfig}
+        __EOF
+
+        # migrate db
+        ${pkgs.php}/bin/php artisan migrate --force
+
+        # clear & create caches (needed in case of update)
+        ${pkgs.php}/bin/php artisan cache:clear
+        ${pkgs.php}/bin/php artisan config:clear
+        ${pkgs.php}/bin/php artisan view:clear
+        ${pkgs.php}/bin/php artisan config:cache
+        ${pkgs.php}/bin/php artisan passport:install
+        #${pkgs.php}/bin/php artisan route:cache
+        #${pkgs.php}/bin/php artisan view:cache
+      '';
+    };
```

Comment:
For the .env file creation, please have a look at the [newest bookstack version](https://github.com/NixOS/nixpkgs/blob/9fc8e0bb56e58f35f5f8bc1323e1325595b4cd89/nixos/modules/services/web-apps/bookstack.nix#L364). It is much cleaner and the proper nix way.

Reply:
Thanks for the hint! I will take a look at it.

Reply:
I have changed the config file generation to match the example from the bookstack NixOS module.